File: Dsa/common/perfectNumber.js
function perfectNumber(num){
    if(num<=1){
        return false
    }
    let def_total=1
    for(let i=2; i<=Math.sqrt(num); i++){
        if(num%i===0){
            def_total += i;
            if(i !==num/i){
                def_total+=num/i
            }
        }
    }
    return def_total===num
}
console.log(perfectNumber(212))

File: Dsa/common/perfectNumber.py
def perfectNumber(num):
    if num<=1: return False

    def_total=1
    for i in range(2, int(num**0.5)+1):
        if num%i==0:
            def_total+=i
            if i != num//i:
                def_total+=num//i
    return def_total==num

print(perfectNumber(212))

def perfect_number(num):
    if num<=1: return False

    def_total=1

    for i in range(2, int(num**0.5)+1):
        if num%i==0:
            def_total+=i
            if i != num//i:
                def_total=num//i
    return def_total==num
print(perfect_number(28))